nemo/collections/asr/models/rnnt_bpe_models.py

```diff
@@ -0,0 +1,263 @@
+
+        # Setup the tokenizer
+        self._setup_tokenizer(tokenizer_cfg)
+
+        # Initialize a dummy vocabulary
+        vocabulary = self.tokenizer.tokenizer.get_vocab()
+
+        joint_config = self.joint.to_config_dict()
+        new_joint_config = copy.deepcopy(joint_config)
+        new_joint_config['vocabulary'] = ListConfig(list(vocabulary.values()))
+        new_joint_config['num_classes'] = len(vocabulary)
+        del self.joint
+        self.joint = EncDecRNNTBPEModel.from_config_dict(new_joint_config)
+
+        decoder_config = self.decoder.to_config_dict()
+        new_decoder_config = copy.deepcopy(decoder_config)
+        new_decoder_config.vocab_size = len(vocabulary)
+        del self.decoder
+        self.decoder = EncDecRNNTBPEModel.from_config_dict(new_decoder_config)
+
+        del self.loss
+        self.loss = RNNTLoss(num_classes=self.joint.num_classes_with_blank - 1)
+
+        if decoding_cfg is None:
+            # Assume same decoding config as before
+            decoding_cfg = self.cfg.decoding
+
+        self.decoding = RNNTBPEDecoding(
+            decoding_cfg=decoding_cfg, decoder=self.decoder, joint=self.joint, tokenizer=self.tokenizer,
+        )
+
+        self.wer = RNNTBPEWER(
+            decoding=self.decoding,
+            batch_dim_index=self.wer.batch_dim_index,
+            use_cer=self.wer.use_cer,
+            log_prediction=self.wer.log_prediction,
+            dist_sync_on_step=True,
+        )
+
+        # Setup fused Joint step
+        if self.joint.fuse_loss_wer:
+            self.joint.set_loss(self.loss)
+            self.joint.set_wer(self.wer)
+
+        # Update config
+        with open_dict(self.cfg.joint):
+            self.cfg.joint = new_joint_config
+
+        with open_dict(self.cfg.decoder):
+            self.cfg.decoder = new_decoder_config
+
+        with open_dict(self.cfg.decoding):
+            self.cfg.decoding = decoding_cfg
+
+        logging.info(f"Changed decoder to output to {self.joint.vocabulary} vocabulary.")
+
+    def _setup_dataloader_from_config(self, config: Optional[Dict]):
+        if 'augmentor' in config:
+            augmentor = process_augmentations(config['augmentor'])
+        else:
+            augmentor = None
+
+        shuffle = config['shuffle']
+
+        # Instantiate tarred dataset loader or normal dataset loader
+        if config.get('is_tarred', False):
+            if ('tarred_audio_filepaths' in config and config['tarred_audio_filepaths'] is None) or (
+                'manifest_filepath' in config and config['manifest_filepath'] is None
+            ):
+                logging.warning(
+                    "Could not load dataset as `manifest_filepath` was None or "
+                    f"`tarred_audio_filepaths` is None. Provided config : {config}"
+                )
+                return None
+
+            shuffle_n = config.get('shuffle_n', 4 * config['batch_size'])
```

Comment:
Should there be some check here to make sure that 4*batch_size is not larger than the dataset? Unlikely to happen, but still a possibility.
In addition, if `shuffle_n` isn't set, should we be checking if `shuffle` exists (and set default to 0 if so)?

Reply:
Hmm getting len(dataset) inside here is pretty hacky, given tarred datasets are iterabledatasets by design. I think its fine to consider 4 * batch size. In the off chance that it exceeds dataset size, the error is probably worth the user setting the value explicitly.

Im a bit confused about second comment, we explicitly set shuffle off in the end of this section. So it has no impact what the user sets it, since inside tarred dataset codepath we switch it off?

Reply:
Re the 4*batch_size: sounds reasonable to me.

About shuffle vs shuffle_n: if I'm remembering correctly, shuffle_n should override shuffle, right? So if we've set shuffle_n to 4*batch_size, then the webdataset created in `get_tarred_bpe_dataset()` will return samples in a shuffled order regardless of `shuffle`. Correct me if I'm wrong, but it seems that we should therefore do something like:
`shuffle_n = config.get('shuffle_n', 4 * config['batch_size'] if shuffle else 0)`
In case the user has not set `shuffle_n` manually to zero, but has set `shuffle=False`.

Reply:
Oh, right. It makes sense to turn off shuffling (even shuffle_n) when using shuffle = False (for train set). I'll add that in my next commit.

Reply:
Done